### PR TITLE
Merge up to 49f3f47b1e9b840e4374d46b105604f4d2c22dd5

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Merge up to https://github.com/facebook/react-native/commit/49f3f47b1e9b840e4374d46b105604f4d2c22dd5 which is a change we had already cherry picked to our repo: https://github.com/microsoft/react-native-macos/pull/1513 . I am doing it this order to try and minimize conflicts

Interesting commits:
- 77e1f83c2835fb1787dbcd83cbc31a6a3044514a Resolve a ViewConfig conflict.
- cc7f31ad171882837ea2fc66ba30b142e0e57b5b  Resolve package.json mismatches.
  - I spoke to @kelset about this one. The RN Core stable branches have extra devDependencies that the main branch does not. We probably had it in our fork as a merge artifact from when we (incorrectly) merged stable branches into main.
- 3ff4b32821c480cbd8a9c7c241c60190789af118 - Stub out references to UITraitCollection on macOS
- cc38d706d0370726adc0a23d70d9f785d34bac38 Resolve a conflict in RCTUITextView. 
  - This was a fix upstream in https://github.com/facebook/react-native/commit/1a83dc36ce0af33ac7a3c311354fce4bfa5ba1a3 . This commits does it's best to mirror that logic in macOS. I tested a few cases and found  it good :)
- ee77d4879e5ef8e4bc9dc9ec45ab10ad8369b8c2 - Refactor some macOS specific ScrollView and TextInput types to add to ViewConfig.

## Changeling

[macOS] [Added] - Merge up to 49f3f47b1e9b840e4374d46b105604f4d2c22dd5

## Test Plan

RNTester launches and seems fine. TextInput (which had a few macOS specific changes) didn't crash for:
- Marked Text cases 
  - typing "b" in Bangla, where it would wait for the next letter
  - typing "dkssud" in 2-Set Korean, where that is replaced with "안녕"
  - Secure Entry

https://user-images.githubusercontent.com/6722175/211929170-d3dfd72b-5d81-4f0e-ae39-8c85541173a5.mov

### CODE REVIEW

**Review by mischreiber — APPROVED** [review]

File: Libraries/Components/TextInput/RCTTextInputViewConfig.js [hunk 1]
```diff
@@ -149,6 +150,15 @@ const RCTTextInputViewConfig = {
     clearTextOnFocus: true,
     showSoftInputOnFocus: true,
     autoFocus: true,
+    ...ConditionallyIgnoredEventHandlers({
```
Comment (Saadnajmi) [inline Libraries/Components/TextInput/RCTTextInputViewConfig.js:153]: Do we need to add more events here?

File: Libraries/Components/ScrollView/ScrollViewNativeComponent.js [hunk 2]
```diff
@@ -138,6 +139,14 @@ const RCTScrollViewViewConfig =
           snapToOffsets: true,
           snapToStart: true,
           zoomScale: true,
+          ...ConditionallyIgnoredEventHandlers({
```
Comment (Saadnajmi) [inline Libraries/Components/ScrollView/ScrollViewNativeComponent.js:142]: Do we need to add more events here?

File: Libraries/NativeComponent/ViewConfigIgnore.js [hunk 3]
```diff
@@ -19,6 +21,30 @@ export function DynamicallyInjectedByGestureHandler<T: {...}>(object: T): T {
...
+ * This annotation also allows us to safely test this removal by setting
+ * global.RN$ViewConfigEventValidAttributesDisabled = true server-side.
+ *
+ * TODO(T110872225): Remove this logic, after achieving platform-consistency
+ */
+export function ConditionallyIgnoredEventHandlers<T: {[name: string]: true}>(
+  value: T,
+): T | void {
+  if (
+    (Platform.OS === 'ios' || Platform.OS === 'macos') &&
```
Comment (Saadnajmi) [inline Libraries/NativeComponent/ViewConfigIgnore.js:40]: Add macOS tag

File: React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm [hunk 4]
```diff
@@ -664,6 +665,7 @@ - (BOOL)_textOf:(NSAttributedString *)newText equals:(NSAttributedString *)oldTe
 #if !TARGET_OS_OSX // [macOS]
   BOOL shouldFallbackToBareTextComparison =
       [_backedTextInputView.textInputMode.primaryLanguage isEqualToString:@"dictation"] ||
+      [_backedTextInputView.textInputMode.primaryLanguage isEqualToString:@"ko-KR"] ||
```
Comment (Saadnajmi) [inline React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm:668]: Port paper fix to here.

File: Libraries/StyleSheet/normalizeColor.js [hunk 5]
```diff
@@ -22,7 +22,7 @@ function normalizeColor(
     const {normalizeColorObject} = require('./PlatformColorValueTypes');
     const normalizedColor = normalizeColorObject(color);
     if (normalizedColor != null) {
-      return color;
+      return normalizedColor;
```
Comment (Saadnajmi) [inline Libraries/StyleSheet/normalizeColor.js:25]: Double check this.

**Pull request merged**